### PR TITLE
interp: always run atomic and volatile loads/stores at runtime

### PR DESCRIPTION
**Pull request by niaow**
This addresses part of the issue in #2409 (but does not completely solve it).

### CODE REVIEW

**Comment by niaow** [comment]
I am still a bit unfamiliar with parts of interp, so I don't know if there is a better way to do this?

**Comment by deadprogram** [comment]
@niaow can you please rebase this against `dev`? Thanks.

File: interp/interpreter.go [hunk 1]
```diff
@@ -928,12 +928,18 @@ func (r *runner) runAtRuntime(fn *function, inst instruction, locals []value, me
 		if inst.llvmInst.IsVolatile() {
 			result.SetVolatile(true)
 		}
+		if ordering := inst.llvmInst.Ordering(); ordering != llvm.AtomicOrderingNotAtomic {
+			result.SetOrdering(ordering)
```
Comment (aykevl) [inline interp/interpreter.go:932]: I think the volatile flag should also be set here (and similarly below for the store instruction).

**Review by aykevl — COMMENTED** [review]
This actually looks reasonable to me. The idea here is that any synchronisation barrier would result in an interp bail out, which seems correct to me (and volatile/atomic operations are some sort of synchronisation barriers).

File: interp/interpreter.go (hunk 1 above)
Comment (niaow) [reply interp/interpreter.go:932]: The volatile flag is already set for volatile loads/stores in the previous if statement. You can see this working in the changed part of `interp/testdata/revert.out.ll`.

**Comment by deadprogram** [comment]
Thanks @niaow now merging.

**Pull request merged**